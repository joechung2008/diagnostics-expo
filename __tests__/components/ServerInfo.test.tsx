import { render } from '@testing-library/react-native';
import ServerInfo from '../../components/ServerInfo';

// Mock the ThemeContext
jest.mock('../../components/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f5f5f5',
      text: '#000000',
      tableBorder: '#cccccc',
      tableHeaderBg: '#e0e0e0',
    },
  }),
}));

describe('ServerInfo', () => {
  const mockServerInfo = {
    hostname: 'test-server.example.com',
    uptime: 1234567890,
    serverId: 'server-123',
    deploymentId: 'deploy-456',
    nodeVersions: 'v18.17.0',
    extensionSync: {
      totalSyncAllCount: 42,
    },
  };

  it('renders all server information correctly', () => {
    const { toJSON } = render(<ServerInfo {...mockServerInfo} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('handles optional fields gracefully', () => {
    const partialServerInfo = {
      hostname: 'test-server.example.com',
      serverId: 'server-123',
      deploymentId: 'deploy-456',
      extensionSync: {
        totalSyncAllCount: 0,
      },
    };

    const { toJSON } = render(<ServerInfo {...partialServerInfo} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
