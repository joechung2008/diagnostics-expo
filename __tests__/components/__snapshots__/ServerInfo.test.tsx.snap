// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerInfo handles optional fields gracefully 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderColor": "#cccccc",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Hostname
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        test-server.example.com
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Uptime
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Server ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        server-123
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Deployment ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        deploy-456
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Node Versions
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Extension Sync | Total Sync All Count
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        0
      </Text>
    </View>
  </View>
</View>
`;

exports[`ServerInfo renders all server information correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderColor": "#cccccc",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Hostname
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        test-server.example.com
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Uptime
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        1234567890
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Server ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        server-123
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Deployment ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        deploy-456
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Node Versions
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        v18.17.0
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Extension Sync | Total Sync All Count
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        42
      </Text>
    </View>
  </View>
</View>
`;
